lib/oli/publishing/delivery_resolver.ex

```diff
@@ -343,6 +349,49 @@ defmodule Oli.Publishing.DeliveryResolver do
     )
   end
 
+  defp hierarchy_nodes_by_sr_id(section, section_resources) do
+    labels =
+      case section.customizations do
+        nil -> CustomLabels.default_map()
+        l -> Map.from_struct(l)
+      end
+
+    Enum.reduce(section_resources, {%{}, nil}, fn sr, {nodes, root} ->
+      node = %HierarchyNode{
+        uuid: uuid(),
+        numbering: %Numbering{
+          index: sr.numbering_index,
+          level: sr.numbering_level,
+          labels: labels
+        },
+        children: sr.children,
+        resource_id: sr.resource_id,
+        revision: %{
+          id: sr.revision_id,
+          resource_id: sr.resource_id,
+          title: sr.title,
+          slug: sr.revision_slug,
+          resource_type_id: sr.resource_type_id,
+          objectives: sr.objectives,
+          graded: sr.graded,
+          relates_to: sr.relates_to,
+          purpose: sr.purpose,
+          deleted: false
+        },
+        project_id: sr.project_id,
+        project_slug: sr.proj_slug,
+        section_resource: sr
+      }
+
+      {
+        Map.put(nodes, sr.id, node),
+        if(section.root_section_resource_id == sr.id, do: node, else: root)
+      }
+    end)
```

Comment:
We should wrap this in a function in order to be able to pipe it into `run()`. And there was a typo in `project_slug: sr.project_slug`

```suggestion
    fn ->
      Enum.reduce(section_resources, {%{}, nil}, fn sr, {nodes, root} ->
        node = %HierarchyNode{
          uuid: uuid(),
          numbering: %Numbering{
            index: sr.numbering_index,
            level: sr.numbering_level,
            labels: labels
          },
          children: sr.children,
          resource_id: sr.resource_id,
          revision: %{
            id: sr.revision_id,
            resource_id: sr.resource_id,
            title: sr.title,
            slug: sr.revision_slug,
            resource_type_id: sr.resource_type_id,
            objectives: sr.objectives,
            graded: sr.graded,
            relates_to: sr.relates_to,
            purpose: sr.purpose,
            deleted: false
          },
          project_id: sr.project_id,
          project_slug: sr.project_slug,
          section_resource: sr
        }

        {
          Map.put(nodes, sr.id, node),
          if(section.root_section_resource_id == sr.id, do: node, else: root)
        }
      end)
    end
```